### PR TITLE
Rest 49: Documentation using Automation Tool DocFX

### PR DESCRIPTION
**Pull request by ujj-s**
# DocFX Tool

- DocFx tool can be used to automate the documentation process and can be used to create good Static Documentation file
- The tool must be installed and configured to the project using `docfx.json` file and the necessary code must be added depending on the class library or .net api project

# Updates

- Added the XML Documentation to the all the code files so that the DocFx can document by generating the HTML files
- The `_site/10.0.2/api` contains all the HTML files and contains a `toc.html` file that contains link to all the files in one

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -125,11 +125,15 @@ csharp_style_pattern_matching_over_is_with_cast_check = true:suggestion
...
+csharp_style_var_for_built_in_types = false:silent
+csharp_style_var_when_type_is_apparent = false:silent
+csharp_style_var_elsewhere = false:silent
 
 # CA2249: Consider using 'string.Contains' instead of 'string.IndexOf'
 dotnet_diagnostic.CA2249.severity = error
+csharp_space_around_binary_operators = before_and_after
+
+# IDE0008: Use explicit type
+dotnet_diagnostic.IDE0008.severity = silent
```
Comment (sameer-s-b) [inline .editorconfig:138]: Please explain the addition of these rules.

File: Word/WordDocumentGenerator.cs [hunk 2]
```diff
@@ -104,6 +113,11 @@ public static void GenerateDocumentByTemplate(string templateFilePath, DocumentD
             }
         }
 
+        /// <summary>
```
Comment (sameer-s-b) [inline Word/WordDocumentGenerator.cs:116]: We should utilize the exception variable and throw it.

File: Word/WordDocumentGenerator.cs [hunk 3]
```diff
@@ -233,35 +276,51 @@ private static void ReplaceImagePlaceholders(string inputFilePath, string output
...
+                        ReplaceImage(mainDocumentPart, drawing, imagePath);
                     }
                 }
             }
 
-            // Overwrite the output file
-            FileStream fileStream = new FileStream(outputFilePath, FileMode.OpenOrCreate, FileAccess.ReadWrite);
-            memoryStream.WriteTo(fileStream);
-            fileStream.Close();
+            // Save the updated document bytes to the output file path
```
Comment (sameer-s-b) [inline Word/WordDocumentGenerator.cs:287]: Are these the changes that were suggested by the editorconfig rules?

**Review by sameer-s-b — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 1 above)
Comment (ujj-s) [reply .editorconfig]: made the changes! Use Explicit Types was disabled somehow, i updated the rule as an error and also made the changes to the code

File: Word/WordDocumentGenerator.cs (hunk 2 above)
Comment (ujj-s) [reply Word/WordDocumentGenerator.cs:116]: made the necessary change

File: Word/WordDocumentGenerator.cs (hunk 3 above)
Comment (ujj-s) [reply Word/WordDocumentGenerator.cs:303]: no, i verified the code with the main branch, made the necessary changes to my repository. My code was not updated with the latest commit.

**Review by sameer-s-b — APPROVED** [review]

**Pull request merged**